### PR TITLE
WiP: SPARK-8064 Upgrade to Hive 1.2

### PR DESCRIPTION
**Pull request by steveloughran**
This is work in progress branch to add Hive 1.2.0 support.

Current status: it compiles, initial instantiations don't throw up obscure linkage errors.  Tests fail fast on ClientWrapper linking
## Build

Some of the POM changes are probably just superstition: I had them in from my original Spark 1.3 + Hive 1.2 build and haven't gone through to see which explicit includes/excludes are needed now that
all the spark dependencies have changed.
## Compile

in `org.apache.spark.sql.hive.client.ClientWrapper`; `loadPartition`,
`loadTable` and `loadDynamicPartitions`: need to choose suitable defaults
for the new options, or add them to the signature

`org.apache.spark.sql.hive.client.ClientWrapper.reset()`: should the `deleteData`
flag be set?
## Kryo versions

HiveShim now has pasted in some of Hive Utilities to ser/deser using
the Spark Kryo version, not Hive's

### CODE REVIEW

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/ClientWrapper.scala [hunk 1]
```diff
@@ -390,7 +390,8 @@ private[hive] class ClientWrapper(
       replace,
       holdDDLTime,
       inheritTableSpecs,
-      isSkewedStoreAsSubdir)
+      isSkewedStoreAsSubdir,
```
Comment (vanzin) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/ClientWrapper.scala]: It's great to see work done on this bug, but I don't think these changes to the `client` sub-package should be part of it. This code needs to work against different Hive versions, so you need something like #6627 here.

There are specific bugs to add support for different metastores (e.g. SPARK-8066, SPARK-8067). There isn't one for 1.2 specifically, so either we should file one, or you can choose to do it as part of this change - but you'll probably want to build on top of my PR for the client part...

**Comment by steveloughran** [comment]
I hadn't seen the other work you are doing. 
how about I worry about hadoop 1.2 & thrift while you get the other bit in, and I can build on that?

**Comment by vanzin** [comment]
Sounds good. I'm mostly waiting for Michael to come back from vacation so he can review that PR.

**Comment by yhuai** [comment]
@steveloughran https://github.com/apache/spark/pull/6627 has been merged. Can you update the PR?

**Comment by marmbrus** [comment]
hey @steveloughran, very excited to hopefully include this feature in Spark 1.5.  Let me know if you are running into any issues with your port.

**Comment by steveloughran** [comment]
I'm going to cancel this and build up a new one, focusing on the hive thriftserver

**Pull request closed without merging**